serializations.md

```diff
@@ -0,0 +1,321 @@
+# Serialization formats
+
+The serialization formats in this document describes the binary format
+used to encode the different objects in the binary format that is used
+for hashing. Other formats may be used for communication between
+nodes.
+
+## Static size object serialization
+
+#### Block header
+
+For block headers all fields sizes are statically known and can be
+constructed directly as a byte array.
+
+| Fieldname | Size (bytes) |
+| --- | --- |
+| height       | 8    |
+| prev_hash    | 32   |
+| txs_hash     | 32   |
+| root_hash    | 32   |
+| target       | 8    |
+| nonce        | 8    |
+| time         | 8    |
+| version      | 8    |
+| pow_evidence | 168  |
+
+
+#### Block
+
+The only difference between a block and its header is the list of
+transactions. The transactions are captured in the header by the
+transaction root hash (txs_hash). The block does not currently have a
+separate binary serialization form since the block hash is computed
+from the block header.
+
+## Dynamic size object serialization
+
+In this section we use `[]` to mean lists, `<name>` to denote fields,
+`,` to separate fields in lists. We use `::` to separate fields from
+their types . We use `int()` to denote the integer type and `binary()`
+to denote the byte array type. Lists are denoted with `[]` in the type
+domain (e.g., `[int()]` is a list of integers). We use `++` as the
+list concatenation operator.
+
+### RLP Encoding
+
+We use the Recursive Length Prefix encodng
+(https://github.com/ethereum/wiki/wiki/RLP) for serializing objects that
+have dynamic sizes of the fields.
+
+RLP ensures that there is only one serialization of each corresponding
+object on the lowest level, but it can only encode two primitive
+objects, lists and byte arrays.
+
+Objects in Æternity are encoded as lists of fields, where the two
+first fields describe the object type and the object version.
+
+```
+[ <tag>, <version>, <field1>, <field2>...]
+```
+
+Since all values are byte arrays in RLP, `int()` needs to be a byte
+array as well. We encode all integers as unsigned, big endian byte
+arrays. To avoid ambiguity in the encoding of integers, we adapt the
+same scheme as RLP and demand that ingegers are encoded with the
+minimal number of bytes (i.e., disallow leading zeroes in the encoded
+byte array).
+
+### Binary serialization
```

Comment:
I understand this is for:
* Objects as in state trees;
* Txs as hashed for txs tree;
* Txs as signed.

Reply:
I added this at the top of the document.